src/adaptors/amqp/qd_connector.c

```diff
@@ -109,54 +114,96 @@ static void try_open_cb(void *context)
 }
 
 
-const qd_server_config_t *qd_connector_config(const qd_connector_t *c)
+/** Close the proton connection
+ *
+ * Scheduled on the target connections thread
+ */
+static void deferred_close(void *context, bool discard)
+{
+    if (!discard) {
+        pn_connection_close((pn_connection_t*)context);
+    }
+}
+
+
+const qd_server_config_t *qd_connector_get_config(const qd_connector_t *c)
 {
-    return &c->config;
+    return &c->ctor_config->config;
 }
 
 
-qd_connector_t *qd_server_connector(qd_server_t *server)
+qd_connector_t *qd_connector(qd_connector_config_t *ctor_config, bool is_data_connector)
```

Comment:
Should this be called `qd_connector_create()` just like `qd_connector_config_create()`

Reply:
Done

Reply:
Please feel free to commit !